src/main/java/com/ciandt/techgallery/service/impl/GooglePlusCommunicationServiceImpl.java

```diff
@@ -0,0 +1,195 @@
+      instance = new GooglePlusCommunicationServiceImpl();
+    }
+    return instance;
+  }
+
+  /*
+   * (non-Javadoc)
+   * 
+   * @see com.ciandt.techgallery.service.GooglePlusCommunicationService#
+   * postGooglePlus(com.ciandt.techgallery.service.enums.FeatureEnum,
+   * java.lang.Boolean, java.lang.String, java.lang.String, java.lang.String,
+   * com.google.appengine.api.users.User, javax.servlet.http.HttpServletRequest)
+   */
+  @Override
+  public void postGooglePlus(FeatureEnum feature, Boolean score, String comment, String currentUserMail,
+      String endorsedMail, String technologyName, String techGalleryLink, User user, HttpServletRequest req)
+          throws InternalServerErrorException, BadRequestException, NotFoundException, IOException {
+
+    validateUser(user);
+    verifyRequirements(user);
+
+    // Create a list of ACL entries
+    PlusDomainsAclentryResource resource = new PlusDomainsAclentryResource();
+    resource.setType("domain");
+
+    List<PlusDomainsAclentryResource> aclEntries = new ArrayList<PlusDomainsAclentryResource>();
+    aclEntries.add(resource);
+
+    Acl acl = new Acl();
+    acl.setItems(aclEntries);
+    acl.setDomainRestricted(true); // Required, this does the domain restriction
+
+    // Create a new activity object to be executed
+    String content = createContent(feature, currentUserMail, endorsedMail, technologyName, score, comment);
+    Activity activity = new Activity().setObject(new Activity.PlusDomainsObject().setOriginalContent(content))
+        .setAccess(acl);
+
+    // Attach the link
+    Activity.PlusDomainsObject.Attachments attachment = new Activity.PlusDomainsObject.Attachments();
+    attachment.setObjectType("article");
+    attachment.setUrl(techGalleryLink);
+    List<Activity.PlusDomainsObject.Attachments> attachments = new ArrayList();
+    attachments.add(attachment); // You can also add multiple attachments to the
+                                 // post
+    activity.getObject().setAttachments(attachments);
+
+    // Creating a google credential in base of the header authorization
+    String header = req.getHeader("Authorization");
+    String accesstoken = header.substring(header.indexOf(' ')).trim();
+    GoogleCredential credential = new GoogleCredential().setAccessToken(accesstoken);
+
+    // Create a new authorized API client according the credential
+    PlusDomains plusDomains = new PlusDomains.Builder(new NetHttpTransport(), new JacksonFactory(), credential).build();
+
+    // Execute the API request, which calls `activities.insert` for the logged
+    // in user
+    activity = plusDomains.activities().insert("me", activity).execute();
+  }
+
+  /**
+   * This method confirm the user preference to post on your Google+ profile
+   * 
+   * @param user
+   *          the user logged in
+   * @throws NotFoundException
+   * @throws BadRequestException
+   * @throws InternalServerErrorException
+   */
+  private void verifyRequirements(User user)
```

Comment:
@tsr-thulio This code is fetching TechGallery twice without need. 'validateUser' method is also fetching TGUser. Try to reuse it.